tests/cmdline/commands/test_presto.py

```diff
@@ -48,3 +48,24 @@ def test_presto(run_cli_command, with_broker, monkeypatch):
             assert profile.process_control_backend == 'core.rabbitmq'
         else:
             assert profile.process_control_backend is None
+
+
+@pytest.mark.usefixtures('empty_config')
+def test_presto_use_postgres(run_cli_command):
+    """Test the ``verdi presto`` with the ``-use-postgres`` flag."""
+    result = run_cli_command(verdi_presto, ['--non-interactive', '--use-postgres'])
+    assert 'Created new profile `presto`.' in result.output
+
+    with profile_context('presto', allow_switch=True) as profile:
+        assert profile.name == 'presto'
+        localhost = Computer.collection.get(label='localhost')
+        assert localhost.is_configured
+        assert profile.storage_backend == 'core.psql_dos'
+
+
+@pytest.mark.usefixtures('empty_config')
+def test_presto_use_postgres_fail(run_cli_command):
```

Comment:
I'd also add a test that successfully utilizes all the available cmdline options.

Reply:
The main point here was to test the exception when the connection failed. I _could_ test the other options, but that would be quite tricky because I would have to start mocking the actual admin username and password of the postgres cluster (and the hostname, as the final option).